### PR TITLE
fix(api): craft workflow runs

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14055.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14057.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14058.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: engine/api/workflow_run_craft.go [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/log"
+	"github.com/ovh/cds/sdk/telemetry"
+	"github.com/pkg/errors"
+	"go.opencensus.io/trace"
+)
+
+func (api *API) WorkflowRunCraft(ctx context.Context, tick time.Duration) error {
+	ticker := time.NewTicker(tick)
+
```
Comment (sguiheux) [inline engine/api/workflow_run_craft.go:20]: close ticker

Comment (fsamin) [reply engine/api/workflow_run_craft.go:21]: done

File: engine/sql/api/213_workflow_run_crafting.sql [hunk 2]
```diff
@@ -0,0 +1,16 @@
+-- +migrate Up
+
+ALTER TABLE "workflow_run" ADD COLUMN to_craft BOOLEAN;
+UPDATE "workflow_run" set to_craft = false;
+ALTER TABLE "workflow_run" ALTER COLUMN to_craft SET DEFAULT false;
+
+ALTER TABLE "workflow_run" ADD COLUMN to_craft_opts JSONB;
+
+UPDATE workflow_run SET to_craft = false WHERE to_craft is null;
```
Comment (yesnault) [inline engine/sql/api/213_workflow_run_crafting.sql:9]: duplicate with line 4 - to delete

File: sdk/error.go [hunk 3]
```diff
@@ -198,6 +198,7 @@ var (
 	ErrUnsupportedMediaType                          = Error{ID: 188, Status: http.StatusUnsupportedMediaType}
 	ErrNothingToPush                                 = Error{ID: 189, Status: http.StatusBadRequest}
 	ErrWorkerErrorCommand                            = Error{ID: 190, Status: http.StatusBadRequest}
+	ErrRepoAnalyzeFailed                             = Error{ID: 192, Status: http.StatusInternalServerError}
```
Comment (yesnault) [inline sdk/error.go:201]: 191 ?

Comment (fsamin) [reply sdk/error.go]: done

File: engine/sql/api/213_workflow_run_crafting.sql (hunk 2 above)
Comment (fsamin) [reply engine/sql/api/213_workflow_run_crafting.sql]: done

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14159.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

File: sdk/workflow_run.go [hunk 4]
```diff
@@ -77,10 +81,29 @@ type WorkflowNodeRunRelease struct {
...
 type WorkflowRunPostHandlerOption struct {
-	Hook        *WorkflowNodeRunHookEvent `json:"hook,omitempty"`
-	Manual      *WorkflowNodeRunManual    `json:"manual,omitempty"`
-	Number      *int64                    `json:"number,omitempty"`
-	FromNodeIDs []int64                   `json:"from_nodes,omitempty"`
+	Hook         *WorkflowNodeRunHookEvent `json:"hook,omitempty"`
+	Manual       *WorkflowNodeRunManual    `json:"manual,omitempty"`
+	Number       *int64                    `json:"number,omitempty"`
+	FromNodeIDs  []int64                   `json:"from_nodes,omitempty"`
+	AuthConsumer *AuthConsumer             `json:"auth_consumer,omitempty"`
```
Comment (richardlt) [inline sdk/workflow_run.go:88]: Looks like its used for event, can we store only an identifiable ?

Comment (fsamin) [reply sdk/workflow_run.go]: done

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14165.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by richardlt — APPROVED** [review]

**Pull request merged**